scene/resources/visual_shader_nodes.cpp

```diff
@@ -581,17 +583,53 @@ Quaternion VisualShaderNodeVec4Constant::get_constant() const {
 	return constant;
 }
 
+void VisualShaderNodeVec4Constant::set_constant_v4(const Vector4 &p_constant) {
+	set_constant(Quaternion(p_constant.x, p_constant.y, p_constant.z, p_constant.w));
+}
+
+Vector4 VisualShaderNodeVec4Constant::get_constant_v4() const {
+	return Vector4(constant.x, constant.y, constant.z, constant.w);
+}
+
+#else
+
+void VisualShaderNodeVec4Constant::set_constant(const Vector4 &p_constant) {
+	if (constant.is_equal_approx(p_constant)) {
+		return;
+	}
+	constant = p_constant;
+	emit_changed();
+}
+
+Vector4 VisualShaderNodeVec4Constant::get_constant() const {
+	return constant;
+}
+
+#endif // DISABLE_DEPRECATED
```

Comment:
We can't do this, the engine shouldn't have a different main behavior based on whether `DISABLE_DEPRECATED` is set.

`DISABLE_DEPRECATED` should just remove conversion code, not change how the engine works for the non-deprecated case.

Reply:
ah, really...